src/Gaufrette/Adapter/AsyncAwsS3.php

```diff
@@ -10,37 +10,24 @@
 /**
  * Amazon S3 adapter using the AsyncAws.
  *
- * @author  Michael Dowling <mtdowling@gmail.com>
+ * @author Michael Dowling <mtdowling@gmail.com>
  * @author Tobias Nyholm <tobias.nyholm@gmail.com>
  */
 class AsyncAwsS3 implements Adapter, MetadataSupporter, ListKeysAware, SizeCalculator, MimeTypeProvider
 {
-    /** @var SimpleS3Client */
-    protected $service;
-    /** @var string */
-    protected $bucket;
-    /** @var array */
-    protected $options;
-    /** @var bool */
-    protected $bucketExists;
-    /** @var array */
-    protected $metadata = [];
-    /** @var bool */
-    protected $detectContentType;
-
-    /**
-     * @param SimpleS3Client $service
-     * @param string   $bucket
-     * @param array    $options
-     * @param bool     $detectContentType
-     */
-    public function __construct(SimpleS3Client $service, $bucket, array $options = [], $detectContentType = false)
-    {
+    protected array $options;
+    protected bool $bucketExists;
+    protected array $metadata = [];
+
+    public function __construct(
+        private readonly SimpleS3Client $service,
```

Comment:
@KevinArtus we said that we will add the property promotion et readonly once the typehint will be merge right ?